WaterLog_Backend/Controllers/SegmentEventsController.cs

```diff
@@ -61,7 +61,32 @@ public async Task AddSegmentEvent([FromBody] SegmentEventsEntry value)
         {
             Procedures proc = new Procedures(_db, _config);
             var ret = await proc.CalculatePeriodWastageAsync(Procedures.Period.Daily);
-            return ret.FirstOrDefault();
+            DataPoints<DateTime, double> v = new DataPoints<DateTime, double>();
+            List<DateTime> dummyHours = new List<DateTime>();
+            for (int i = 0; i < 24; i++)
+            {
+                DateTime tempTime = DateTime.Now;
+                DateTime returnV = new DateTime(tempTime.Year, tempTime.Month, tempTime.Day, i, 0, 0);
+                for (int j =0; j< ret.Count(); j++)
+                {
+                    List<DateTime> t = ret.ElementAt(j).getv2();
+                    List<double> d = ret.ElementAt(j).getv();
+                    if (t.ElementAt(0).Hour == returnV.Hour)
+                    {
+                        v.AddPoint(t.ElementAt(0), d.ElementAt(0));
+                    }
+                    else
+                    {
+                        v.AddPoint(t.ElementAt(0), 0);
+                    }
+
+                }
+               
+
+                
+            }
+
+            return v;
```

Comment:
Can we please name these variables better? Without inspecting the code, I'll have no way of knowing what v, t, or d represents or what the difference is between ret and returnV

Reply:
changed